src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -406,24 +407,42 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
+        report = []
+        reqArgsDiff = deepcopy(request_args)
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return reduceReport(report)
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgsDiff)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
+                workload.setPriority(reqArgsDiff['RequestPriority'])
+                workload.saveCouchUrl(workload.specUrl())
+                cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgsDiff['RequestPriority']))
+            elif workqueue_stat_validation(reqArgsDiff):
+                report.append(self.reqmgr_db_service.updateRequestStats(workload.name(), reqArgsDiff))
+                cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgsDiff))
+            elif reqArg == "SiteWhitelist":
+                workload.setSiteWhitelist(reqArgsDiff["SiteWhitelist"])
+                cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteWhitelist']))
+            elif reqArg == "SiteBlacklist":
+                workload.setSiteBlacklist(reqArgsDiff["SiteBlacklist"])
+                cherrypy.log('Updated SiteBlacklist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteBlacklist']))
+            else:
+                reqArgsNothandled.append(reqArg)
+                cherrypy.log("Uunhandled argument for no-status update: %s" % reqArg)
+
+        if reqArgsNothandled:
+            msg = "There were unhandled arguments left for no-status update: %s" % reqArgsNothandled
             raise InvalidSpecParameterValue(msg)
 
-        return report
+        report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
```

Comment:
Note that if you rollback your change above - about raise an exception for unsupported parameters - that you also need to change `reqArgsDiff` here, otherwise you would be updating unsupported properties as well.